### PR TITLE
[TS] LPS-146970 BackgroundTaskManager initialization

### PR DESCRIPTION
**Pull request by rafaprax**

### CODE REVIEW

**Comment by tinatian** [comment]
@rafaprax Nice job!
@catarineaguiar Please help me scan test results.

Thanks!
Tina.

**Comment by rafaprax** [comment]
ci:test:sf

**Comment by catarineaguiar** [comment]
@rafaprax @tinatian no regressions were found!

**Comment by tinatian** [comment]
@shuyangzhou This is good to go.

Tina.

File: modules/apps/portal-background-task/portal-background-task-service/src/main/java/com/liferay/portal/background/task/internal/BackgroundTaskMessagingConfigurator.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	protected void deactivate() {
+		for (ServiceRegistration<Destination> serviceRegistration :
+				_serviceRegistrations) {
+
+			Destination destination = _bundleContext.getService(
+				serviceRegistration.getReference());
+
+			serviceRegistration.unregister();
+
+			destination.destroy();
```
Comment (shuyangzhou) [inline modules/apps/portal-background-task/portal-background-task-service/src/main/java/com/liferay/portal/background/task/internal/BackgroundTaskMessagingConfigurator.java:86]: @rafaprax Destination is opened/closed by DefaultMessageBus upon register/unregister the service, you should not do it again here.

File: modules/apps/portal-background-task/portal-background-task-service/src/main/java/com/liferay/portal/background/task/internal/BackgroundTaskMessagingConfigurator.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	@Reference
+	private DestinationFactory _destinationFactory;
+
+	@Reference
+	private LockManager _lockManager;
+
+	@Reference
+	private MessageBus _messageBus;
+
+	private final Set<ServiceRegistration<Destination>> _serviceRegistrations =
```
Comment (shuyangzhou) [inline modules/apps/portal-background-task/portal-background-task-service/src/main/java/com/liferay/portal/background/task/internal/BackgroundTaskMessagingConfigurator.java:138]: @rafaprax this supposed to keep track of ServiceRegistration for unregistering, there can not be any duplication on ServiceRegistrations, this should be a plain List for better performance

**Comment by shuyangzhou** [comment]
@rafaprax see above comments, please fix and resend

**Comment by shuyangzhou** [comment]
ci:close

**Comment by rafaprax** [comment]
resent here https://github.com/shuyangzhou/liferay-portal/pull/10618

**Pull request closed without merging**